### ISSUE TITLE
Make get(), put(), delete() return proper builders

### ISSUE THREAD

**Issue by Mallets** [eclipse-zenoh/zenoh#239]
`subscribe()`, `queryable()`, and `publish()` return respectively `SubscriberBuilder`, `QueryableBuilder`, `PublisherBuilder`.

For consistency, `get()`, `put()`, `delete()` should also return `GetBuilder`, `PutBuilder`, `DeleteBuilder` types. 
This is particular relevant for the different types returned by `put()` and `delete()` since some configs used in `put()` operations don’t make sense for `delete()` ones.

Tasks:

- [x] GetBuilder
- [x] PutBuilder
- [x] DeletBuilder

**Comment by Mallets** [comment]
https://github.com/eclipse-zenoh/zenoh/pull/255 has been merged into api-changes branch. 
This issue will be automatically closed once api-changes is merged into master.

**Comment by Mallets** [comment]
Merged into master.
